lib/src/features/leaderboard/ui/leaderboard_widget.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:lichess_mobile/src/features/leaderboard/data/leaderboard_repository.dart';
+import 'package:lichess_mobile/src/widgets/feedback.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/common/lichess_icons.dart';
+
+import '../model/leaderboard.dart';
+import './leaderboard_screen.dart';
+
+final leaderListProvider = FutureProvider.autoDispose((ref) async {
+  final leaderRepo = ref.watch(leaderboardRepositoryProvider);
+  final either = await leaderRepo.getLeaderboard().run();
+  return either.match((error) => throw error, (data) {
+    ref.keepAlive();
```

Comment:
don't use `ref.keepAlive()` as it will cache forever the data (sorry for the example in my code where I used it wrongly too...);

The leaderboard data don't change often so we may later add a more clever mechanism to cache the data but with a timeout when it invalidates itself. 
Perhaps later we'll also implement a pull to refresh mechanism on the home page. But let's not add unnecessary complexity for now.